src/main/java/randoop/main/GenInputsAbstract.java

```diff
@@ -121,6 +121,36 @@ protected GenInputsAbstract(
   @Option("File that lists methods under test")
   public static Path methodlist = null;
 
+  /**
+   * Automatically add dependencies to the classes under test. For any classes, methods, and
+   * constructors specified by the user, this option adds their argument classes. (The user
+   * specifies code to test using the {@code --testjar}, {@code --classlist}, {@code --testclass},
+   * and {@code --methodlist} options.)
+   *
+   * <p>By default, transitive dependencies are not included. The {@code
+   * --test-add-dependencies-depth} option includes transitive dependencies.
+   *
+   * <p>Dependencies that are interfaces or abstract classes are not added. You may need to
+   * additionaly specify concrete classes.
+   *
+   * <p>When {@code --test-add-dependencies=true}, it is recommended to use the {@code
+   * --require-covered-classes} option with the same classes, to avoid outputting tests of
+   * dependencies to test suites.
+   */
+  @Option("Add classes that are method/constructor arguments")
+  public static boolean test_add_dependencies = false;
+
+  /**
+   * Configures the depth of adding dependendencies when using the {@code --test-add-dependencies}
+   * option. When this option is 1, only the direct dependencies of tested classes and methods are
+   * added. When this option is 2, dependencies of dependendencies are also added. When this option
+   * is 3, dependencies of dependendencies of dependendencies are also added. And so forth.
+   *
+   * <p>If code contains circular dependencies, Randoop will add those dependencies repeatedly.
```

Comment:
When we call method `GenTests.getDependentClassnamesFromClassnamesWithDepth` recursively, we use dependencies classnames created inside the method instead of using classnames provided to method in arguments. So when searching dependencies depth X, we only know dependencies depth X-1, but not previous. So in situation when tested class A uses B and B uses A again (this code has a high coupling and is a red flag of bad architecture, but unfortunately such situations are still possible) and we search for dependencies depth 2 we only know about class B, not A. In the end we will have both classes A and B, but searching for the same classes will cost a lot of extra resources.

Reply:
I agree that from an implementation point of view, Randoop will notice classes repeatedly.

I think that from a user point of view, the classes only appear once, as if the user had specified them in `--testclass` without duplication.  (Am I right about that?)

Since this Javadoc is user documentation, I would drop this remark.  You could add it as a code comment, where it will be visible only to people working on the Randoop source code (the main people who should care).

Reply:
Also, my change to the `test_add_dependencies` loop may mean that the comment isn't true even from an implementation standpoint.  (Please see my changes and let me know if they look wrong.)